DragaliaAPI.Database/Repositories/FortRepository.cs

```diff
@@ -58,4 +59,64 @@ public async Task<bool> CheckPlantLevel(FortPlants plant, int requiredLevel)
 
         return result;
     }
+
+    public async Task GetFortPlantIdList(IEnumerable<int> fort_plant_id_list)
+    {
+        // What do
+    }
+
+    public async Task<bool> InitFortDetail(string accountId)
+    {
+        await apiContext.PlayerFortDetails.AddAsync(
+            new DbFortDetail()
+            {
+                DeviceAccountId = accountId,
+                CarpenterNum = 2,
+                MaxCarpenterCount = 5,
+                WorkingCarpenterNum = 0
+            }
+        );
+        await apiContext.SaveChangesAsync();
+        return true;
+    }
+
+    public async Task UpdateFortCarpenterNum(string accountId, int carpenterNum)
+    {
+        DbFortDetail fortDetail = await apiContext.PlayerFortDetails
+            .Where(x => x.DeviceAccountId == accountId)
+            .FirstAsync();
+        fortDetail.CarpenterNum = carpenterNum;
+        apiContext.Entry(fortDetail).State = EntityState.Modified;
+    }
+
+    public async Task UpdateFortWorkingCarpenter(string accountId, int working_carpenter_num)
+    {
+        DbFortDetail fortDetail = await apiContext.PlayerFortDetails
+            .Where(x => x.DeviceAccountId == accountId)
+            .FirstAsync();
+        fortDetail.WorkingCarpenterNum = working_carpenter_num;
+        apiContext.Entry(fortDetail).State = EntityState.Modified;
+    }
+
+    public async Task<DbFortBuild> GetBuilding(string accountId, long buildId)
+    {
+        return await apiContext.PlayerFortBuilds
+            .Where(x => x.DeviceAccountId == accountId && x.BuildId == buildId)
+            .FirstAsync();
```

Comment:
this will throw an exception if the player tries to request a build ID they don't own -- unlikely to happen but maybe worth adding handling for?